### ISSUE TITLE
[BUG] using "compose" as --project-name makes compose binary error

### ISSUE THREAD

**Issue by xAt0mZ** [docker/compose#10647]
### Description

When using `compose` as project name, with `--project-name` option, compose binary errors.

**Actual**
```
>$ ./docker-compose --project-name test down --remove-orphans
Warning: No resource found to remove for project "test".

>$ ./docker-compose --project-name compose down --remove-orphans
unknown flag: --remove-orphans
```

**Expected**
```
>$ ./docker-compose --project-name test down --remove-orphans
Warning: No resource found to remove for project "test".

>$ ./docker-compose --project-name compose down --remove-orphans
Warning: No resource found to remove for project "compose".
```

Tested with multiple "docker related" keywords and only `compose` produces this error.

### Compose Version

```Text
Tested all of the following versions

Docker Compose version 2.17.3
Docker Compose version 2.17.2
Docker Compose version 2.17.0
```

### Docker Environment

```Text
Multiple docker versions / environments setups tested so unrelated

Docker versions
23.0.5
19.03.12

Environments
Archlinux bare metal
boot2docker docker-machine
Ubuntu (stable) virtualbox VM
```

**Comment by ndeloof** [comment]
Tried to reproduce without success with latest release:
```
$ docker compose --project-name compose down --remove-orphans
Warning: No resource found to remove for project "compose".
$ docker-compose --project-name compose down --remove-orphans
Warning: No resource found to remove for project "compose".
```

**Comment by xAt0mZ** [comment]
`docker compose` works but it won't solve the issue in my usage scenario.

Also tested with the latest version and I'm still able to reproduce :arrow_down: 
* download the latest version with `wget -O /tmp/docker-compose https://github.com/docker/compose/releases/download/v2.18.1/docker-compose-linux-x86_64`
* make the binary executable `chmod +x  /tmp/docker-compose`
* try `/tmp/docker-compose --project-name compose down --remove-orphans`
  * fail with `unknown flag: --remove-orphans`
* try `/tmp/docker-compose --project-name test down --remove-orphans`
  * ok with `Warning: No resource found to remove for project "test".`

The TLDR of usage scenario: at [portainer/portainer](https://github.com/portainer/portainer) we download & ship compose with our containers (so we can guarantee it is always located somewhere we can use it)
All of our commands are generated to use our shipped binary
We can't rely on `docker compose` because we don't invoke docker binary directly ;we use docker API to communicate with the env (there is no API for compose actions)

**Comment by xAt0mZ** [comment]
Another piece of info. Considering
```
$> docker info
Client:
 Plugins:
  compose: Docker Compose (Docker Inc.)
    Version:  2.17.3
    Path:     /usr/lib/docker/cli-plugins/docker-compose
```
and
```sh
$> which docker-compose
/usr/bin/docker-compose
```
Directly using any of those doesn't work, using `docker compose` works. Idk which one `docker compose` uses, but there is something going on docker side to make it work?
